src/SFA.DAS.AssessorService.Web/Controllers/ApplyForWithdrawalController.cs

```diff
@@ -46,93 +55,74 @@ public class ApplyForWithdrawalController : AssessorController
-                {
-                    return RedirectToAction(nameof(ChooseStandardForWithdrawal), nameof(ApplyForWithdrawalController).RemoveController());
-                }
+                return RedirectToRoute(CheckWithdrawalRequestRouteGet, new { backRouteName = TypeofWithdrawalRouteGet });
+            }
+            else if (viewModel.TypeOfWithdrawal == ApplicationTypes.StandardWithdrawal)
+            {
+                return RedirectToRoute(ChooseStandardForWithdrawalRouteGet);
             }
 
-            return RedirectToAction(nameof(TypeOfWithdrawal), nameof(ApplyForWithdrawalController).RemoveController());
+            return RedirectToRoute(DashboardController.DashboardIndexRouteGet);
         }
 
-        [HttpGet]
+        [HttpGet("withdrawal/choose-standard", Name = ChooseStandardForWithdrawalRouteGet)]
         [ModelStatePersist(ModelStatePersist.RestoreEntry)]
         [TypeFilter(typeof(MenuFilter), Arguments = new object[] { Pages.Dashboard })]
         public async Task<IActionResult> ChooseStandardForWithdrawal(int? pageIndex)
         {
-            return RedirectToAction("Index", "Dashboard");
-
             var contact = await GetUserContact();
             var org = await _orgApiClient.GetEpaOrganisationById(contact.OrganisationId?.ToString());
 
-            var modelStateSelectedStandardForWithdrawal =
-                int.TryParse(
-                    ModelState[nameof(ChooseStandardForWithdrawalViewModel.SelectedStandardForWithdrawal)]?.AttemptedValue,
-                    out int result)
-                ? result
-                : (int?)null;
-
             var applications = await GetWithdrawalApplications(contact.Id);
             var standards = await _standardsApiClient.GetEpaoRegisteredStandards(org.OrganisationId, pageIndex ?? 1, 10);
 
             var viewModel = new ChooseStandardForWithdrawalViewModel()
             {
-                SelectedStandardForWithdrawal = ModelState.IsValid ? null : modelStateSelectedStandardForWithdrawal,
                 Standards = standards.Convert(x => new RegisteredStandardsViewModel()
                 {
                     StandardName = x.StandardName,
                     Level = x.Level,
                     ReferenceNumber = x.ReferenceNumber,
-                    NumberOfVersions = x.NumberOfVersions,
+
+                    // find a previous withdrawal application for the same reference number, if there is
+                    // one then this is used to redirect the user to the previous withdrawal for the same
+                    // standard this will prevent a withdrawal from the same standard twice i.e. it will
+                    // prevent the withdrawal from a reapply for the same standard which most likely was
+                    // not the intention see QF-1608
                     ApplicationId = applications.FirstOrDefault(a => !string.IsNullOrWhiteSpace(a.StandardReference) &&
```

Comment:
SonarDaddy says `Find()` is at least twice as fast as `FirstOrDefault()` for Lists

Reply:
Hmmm - I don't think I want to change that